tools/walletextension/wallet_extension.go

```diff
@@ -74,6 +74,15 @@ func (w *WalletExtension) ProxyEthRequest(request *accountmanager.RPCRequest, co
 	// proxyRequest will find the correct client to proxy the request (or try them all if appropriate)
 	var rpcResp interface{}
 
+	// check if the requested method is GetStorageAt, since we use for gettingUserID if correct parameters are used
+	if request.Method == rpc.GetStorageAt {
+		interceptedResponse := w.getStorageAtInterceptor(request, hexUserID)
+		if interceptedResponse != nil {
+			w.logger.Info("interception successful for getStorageAt, returning userID response")
+			return interceptedResponse, nil
```

Comment:
nit: use the short form

```suggestion
		if w.getStorageAtInterceptor(request, hexUserID) != nil {
			w.logger.Info("interception successful for getStorageAt, returning userID response")
			return interceptedResponse, nil
```